lib/purescript-cst/src/Language/PureScript/AST/Declarations.hs

```diff
@@ -513,6 +522,26 @@ declName KindDeclaration{} = Nothing
 declName TypeDeclaration{} = Nothing
 declName RoleDeclaration{} = Nothing
 
+-- | Same as @declName@ except that KindDeclaration's names
+-- are also included
+declDocName :: Declaration -> Maybe Name
+declDocName (DataDeclaration _ _ n _ _) = Just (TyName n)
+declDocName (TypeSynonymDeclaration _ n _ _) = Just (TyName n)
+declDocName (ValueDeclaration vd) = Just (IdentName (valdeclIdent vd))
+declDocName (ExternDeclaration _ n _) = Just (IdentName n)
+declDocName (ExternDataDeclaration _ n _) = Just (TyName n)
+declDocName (FixityDeclaration _ (Left (ValueFixity _ _ n))) = Just (ValOpName n)
+declDocName (FixityDeclaration _ (Right (TypeFixity _ _ n))) = Just (TyOpName n)
+declDocName (TypeClassDeclaration _ n _ _ _ _) = Just (TyClassName n)
+declDocName (TypeInstanceDeclaration _ _ _ n _ _ _ _) = IdentName <$> hush n
+declDocName (KindDeclaration _ _ n _) = Just (TyName n)
```

Comment:
I think this is why type class' kind signatures don't appear in the final docs whereas data, type, and newtype do. The returned value for `KindDeclaration`s here is `TyName`, which matches the one for `DataDeclaration` and `TypeSynonymDeclaration`. However, it doesn't match the returned value for `TypeClassDeclaration`.

Reply:
This is indeed the issue. I'll push a few more commits and then this PR will be ready for review.